sound/soc/sof/ipc4-topology.c

```diff
@@ -1044,6 +1056,37 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	if (ret < 0)
 		return ret;
 
+	switch (swidget->id) {
+	case snd_soc_dapm_dai_in:
+	case snd_soc_dapm_dai_out:
+	{
+		switch (ipc4_copier->dai_type) {
+		case SOF_DAI_INTEL_ALH:
+		{
+			struct sof_ipc4_alh_configuration_blob *blob;
+			u32 ch_map;
+			int i;
+
+			blob = (struct sof_ipc4_alh_configuration_blob *)ipc4_copier->copier_config;
+			/* TODO: add aggregation mode support */
+			blob->alh_cfg.count = 1;
+			blob->alh_cfg.mapping[0].alh_id = copier_data->gtw_cfg.node_id;
+			blob->gw_attr.lp_buffer_alloc = 0;
+			copier_data->gtw_cfg.config_length = sizeof(*blob) >> 2;
+
+			/* Get channel_mask from ch_map */
+			ch_map = copier_data->base_config.audio_fmt.ch_map;
+			for (i = 0; ch_map > 0; i++) {
+				if ((ch_map & 0xf) != 0xf)
+					 blob->alh_cfg.mapping[0].channel_mask |= BIT(i);
+				ch_map >>= 4;
+			}
+			break;
+		}
```

Comment:
We set ALH blob here after `sof_ipc4_init_audio_fmt()` because we need `copier_data->base_config.audio_fmt.ch_map`, other dai types are handled by `snd_sof_get_nhlt_endpoint_data()` before `sof_ipc4_init_audio_fmt()`.
I am thinking is it better to use `if (ipc4_copier->dai_type == SOF_DAI_INTEL_ALH)` instead of `switch case`?

Reply:
Changed to if and added comment to explain why we set ALH blob here.